src/VisualStudio/Core/Def/Implementation/Remote/ServiceHubRemoteHostClient.cs

```diff
@@ -210,12 +218,32 @@ private void UnregisterGlobalOperationNotifications()
             localTask.Wait();
         }
 
+        private async Task RpcInvokeAsync(string targetName, params object[] arguments)
+        {
+            // handle exception gracefully. don't crash VS due to this.
+            // especially on shutdown time. because of pending async BG work such as 
+            // OnGlobalOperationStarted and more, we can get into a situation where either
+            // we are in the middle of call when we are disconnected, or we runs
+            // after shutdown.
+            try
+            {
+                await _rpc.InvokeWithCancellationAsync(targetName, arguments, _shutdownCancellationTokenSource.Token).ConfigureAwait(false);
+            }
+            catch (Exception ex) when (ReportUnlessCanceled(ex))
+            {
+                if (!_shutdownCancellationTokenSource.IsCancellationRequested)
+                {
+                    RemoteHostCrashInfoBar.ShowInfoBar(Workspace);
+                }
+            }
+        }
+
         private void OnGlobalOperationStarted(object sender, EventArgs e)
         {
             lock (_globalNotificationsGate)
             {
                 _globalNotificationsTask = _globalNotificationsTask.ContinueWith(
-                    continuation, CancellationToken.None, TaskContinuationOptions.None, TaskScheduler.Default).Unwrap();
+                    continuation, _shutdownCancellationTokenSource.Token, TaskContinuationOptions.None, TaskScheduler.Default).Unwrap();
```

Comment:
:question: This omits `LazyCancellation`. Is there a reason why we cannot use that here (e.g. via `SafeContinueWith`)?

Reply:
sure.